src/views/Home.jsx

```diff
@@ -1,12 +1,82 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList.jsx';
+import { useState, useEffect } from 'react';
+import { createList } from '../api/firebase.js';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, userId, userEmail, setListPath }) {
+	const [newListName, setNewListName] = useState('');
+	const [newListSuccess, setNewListSuccess] = useState(null);
+	const [message, setMessage] = useState(null);
+
+	const navigate = useNavigate();
+
+	const handleChange = (e) => {
+		setNewListName(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const currentLists = data.map((list) => {
+			return list.name;
+		});
+
+		if (newListName.includes('/')) {
+			alert(
+				'Invalid list name. "/" is an unsupported character, please try a different list name.',
+			);
+			return;
+		}
+		//Check if the new list has a title that already exists in db
+		else if (currentLists.indexOf(newListName) > -1) {
+			alert('List already exists. Please enter a unique list name.');
+		} else {
+			//Create new list
+			const response = await createList(userId, userEmail, newListName);
+			//If list creation successful, set states
+			if (response) {
+				setNewListSuccess(true);
+				setMessage('List created successfully.');
+				//If list creation error, set states
+			} else {
+				setNewListSuccess(false);
+				setMessage('List could not be created.');
+			}
+		}
+
+		await setListPath(`${userId}/${newListName}`);
```

Comment:
The `await` is not required here because `setListPath` is not an async function.

![CleanShot 2024-02-15 at 18 16 52](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/22878284/2e09ddda-ba2b-4413-9a5d-fccf44c95fc9)

Reply:
👍🏻